codex-cli/src/utils/check-in-vcs.ts

```diff
@@ -29,3 +29,45 @@ export function checkInGit(workdir: string): boolean {
     return false;
   }
 }
+
+/**
+ * Returns true if the given directory is part of a Mercurial (hg) repository.
+ *
+ * Uses the canonical Mercurial command `hg root` which exits with status 0
+ * when executed anywhere inside a working tree (including the repo root) and
+ * exits with a non-zero status otherwise. We only rely on the exit code.
+ */
+export function checkInHg(workdir: string): boolean {
+  try {
+    execSync("hg root", { cwd: workdir, stdio: "ignore" });
```

Comment:
In general, we prefer `spawn(Array<string>)` to be extra safe, though I admit there is no string interpolation in this case. But it's still good to stick with `spawn()` since folks tend to copypasta patterns they see.